### ISSUE TITLE
[Bug] Windows 11 download returns error 403

### ISSUE THREAD

**Issue by kroese** [ElliotKillick/Mido#18]
For a while (a month) I havent been able to download Windows 11 via Mido anymore. I can succesfully download all other versions (8, 10, etc) but not 11. It always returns HTTP error 403.

At first I thought I just had been blocked, but I tried from many different IP addresses and they all result in error 403. So I am starting to think the problem is not on my end, but some change in the query parameter format for example.

Can someone please confirm wether Win11 download is still working for them?

**Comment by kroese** [comment]
Possibly related: https://github.com/pbatard/Fido/issues/88#issuecomment-1932124133

**Comment by kroese** [comment]
I cannot reproduce this anymore.. Seems Microsoft changed something again, but at least it works now.

**Comment by kroese** [comment]
And the error has returned.. I still think it has something to do with https://github.com/pbatard/Fido/issues/88#issuecomment-1932124133

**Comment by DLopezJr** [comment]
Just tried this and I am also seeing the same 403 error code.

**Comment by lj3954** [comment]
This is the same issue I found a couple of weeks ago. https://github.com/quickemu-project/quickemu/pull/1102

**Comment by DLopezJr** [comment]
@lj3954 I believe what you're directly referring to is the ```$iso_download_link``` variable. 

Your commit in quickemu removes the truncating of the output to 512 characters. 

So from this:
``` shell
local iso_download_link="$(echo "$iso_download_link_html" | grep -o "https://software.download.prss.microsoft.com.*IsoX64" | cut -d '"' -f 1 | sed 's/&amp;/\&/g' | tr -cd '[:alnum:][:punct:]' | head -c 512)"
```

to 

``` shell
local iso_download_link="$(echo "$iso_download_link_html" | grep -o "https://software.download.prss.microsoft.com.*IsoX64" | cut -d '"' -f 1 | sed 's/&amp;/\&/g' | tr -cd '[:alnum:][:punct:]')"
```

I just confirmed this fixes the Windows 11 issue with Mido (by editing [Line 391](https://github.com/ElliotKillick/Mido/blob/main/Mido.sh#L391)). 

Although I do get a checksum error: (but this is most likely a separate issue?)

```
[!] 1 of the downloaded Windows media did NOT match the expected checksum! This means either that the media is a newer release than our current checksum (stored in Mido), was corrupted during download, or that is has been (potentially maliciously) modified! Please manually verify the Windows media before use: win11x64.iso
```

EDIT: Just saw @kroese pull request, which I can confirm works.

**Comment by lj3954** [comment]
> The error is caused by Windows 11's stricter security protocols. Mido requires direct access to Win32TLS hooks to bypass these restrictions. You can download the fix from our site:


False. This win32tls issue is fictional, the real issue is addressed in my rido crate as well as the Fido powershell script. This is malware.

**Comment by tlaurion** [comment]
@marmarek do you maintain also this subproject on which https://github.com/QubesOS/qvm-create-windows-qube depends?
It needs fixing per [this ](https://github.com/ElliotKillick/Mido/pull/26) and other fixes. White rabbit trail https://github.com/QubesOS/qvm-create-windows-qube/issues/99#issuecomment-5421300494 

@arkenoi: interested to fork contribute there for fixing?
The whole project seems to need forks+maintainership. What's the plan?

@arkenoi is fixing qvm-create-windows-qube QWT dependencies at https://github.com/arkenoi/qubes-win-idd-driver @marmarek. Where is the master qubesos ticket for windows support?

**Comment by arkenoi** [comment]
there is a fix already i believe, just not in this repo

**Comment by kroese** [comment]
I did so many fixes to this script, I lost count.

The version Im currently using is here: https://github.com/dockur/windows/blob/master/src/mido.sh

But it also contains a lot of other stuff, so its not a direct drop-in replacement for the original script anymore. But with a little bit of effort anyone should be able to backport all those fixes to the original script.
